src/emmetHelper.ts

```diff
@@ -38,29 +38,68 @@ const defaultVendorProperties = {
 /**
  * Emmet configuration as derived from the Emmet related VS Code settings
  */
-export interface EmmetConfiguration {
-	showExpandedAbbreviation?: string;
-	showAbbreviationSuggestions?: boolean;
-	syntaxProfiles?: object;
-	variables?: object;
-	preferences?: object;
-	excludeLanguages?: string[];
-	showSuggestionsAsSnippets?: boolean;
+type VscodeEmmetConfiguration = {
+    excludeLanguages?: any[],
+    extensionsPath?: string,
+    includeLanguages?: Record<string, string>,
+    optimizeStylesheetParsing?: boolean,
+    preferences?: {
+		"bem.elementSeparator"?: string,
+		"bem.modifierSeparator"?: string,
+		"css.floatUnit"?: string,
+		"css.fuzzySearchMinScore"?: number,
+		"css.intUnit"?:string,
+		"css.mozProperties"?: string,
+		"css.msProperties"?:string,
+		"css.oProperties"?:string,
+		"css.propertyEnd"?: string,
+		"css.valueSeparator"?:string,
+		"css.webkitProperties"?: string,
+		"filter.commentAfter"?: string,
+		"filter.commentBefore"?: string,
+		"filter.commentTrigger"?: string[],
+		"format.forceIndentationForTags"?: string,
+		"format.noIndentTags"?: string,
+		"profile.allowCompactBoolean"?: boolean,
+		"sass.propertyEnd"?: string,
+		"sass.valueSeparator"?: string,
+		"stylus.propertyEnd"?: string,
+		"stylus.valueSeparator"?: string,
+
+		// Why following aren't in the json schema but is in tests?
```

Comment:
I picked these from the json schema but looks like there are some missing. Probably will have to dig the source code of old emmet to get the exact types. I guess it would be fine to not be perfect here because  if it's not in the schema the users probably aren't using it. And the compensation will be done when the schema reflects the newer versions options.